File: generate_router.dart
import 'dart:io';

void main() {
  final directories = [
    'e:/Evoeline/lib/screens',
    'e:/Evoeline/lib/screens1-22',
  ];
  final imports = <String>[];
  final routes = <String>[];

  final classRegex = RegExp(r'class\s+(Screen[a-zA-Z0-9_]+)\s+extends');
  final routeNameRegex = RegExp(r'Screen\d+(.*)');

  for (final dirPath in directories) {
    final dir = Directory(dirPath);
    if (!dir.existsSync()) continue;

    for (final entity in dir.listSync()) {
      if (entity is File && entity.path.endsWith('.dart')) {
        final content = entity.readAsStringSync();
        final match = classRegex.firstMatch(content);
        if (match != null) {
          final className = match.group(1)!;
          final filename = entity.uri.pathSegments.last;
          final relDir = dirPath.contains('screens1-22')
              ? 'screens1-22'
              : 'screens';
          imports.add("import '../$relDir/$filename';");

          final routeNameMatch = routeNameRegex.firstMatch(className);
          String routeName;
          String routePath;

          if (routeNameMatch != null && routeNameMatch.group(1)!.isNotEmpty) {
            final namePart = routeNameMatch.group(1)!;
            routeName = namePart[0].toLowerCase() + namePart.substring(1);
            routePath = '/$routeName';
            if (routeName.toLowerCase().contains('splash') ||
                className == 'Screen1Splash') {
              routePath = '/';
              routeName = 'splash';
            }
          } else {
            routeName = className.toLowerCase();
            routePath = '/$routeName';
          }

          routes.add('''
      GoRoute(
        path: '$routePath',
        name: '$routeName',
        builder: (context, state) => const $className(),
      ),''');
        }
      }
    }
  }

  final template =
      """
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';

${imports.join('\n')}

final routerProvider = Provider<GoRouter>((ref) {
  return GoRouter(
    initialLocation: '/',
    debugLogDiagnostics: true,
    routes: [
${routes.join('')}
    ],
  );
});
""";

  File('e:/Evoeline/lib/router/app_router.dart').writeAsStringSync(template);
  print('Generated ${routes.length} routes.');
}

File: generate_router.py
import os
import re

directories = ['e:/Evoeline/lib/screens', 'e:/Evoeline/lib/screens1-22']
imports = []
routes = []

for idx, directory in enumerate(directories):
    if not os.path.exists(directory): continue
    for filename in os.listdir(directory):
        if not filename.endswith('.dart'): continue
        filepath = os.path.join(directory, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            match = re.search(r'class\s+(Screen[a-zA-Z0-9_]+)\s+extends', content)
            if match:
                class_name = match.group(1)
                rel_dir = 'screens' if 'screens/' in filepath.replace('\\', '/') else 'screens1-22'
                imports.append(f"import '../{rel_dir}/{filename}';")
                
                route_name_match = re.search(r'Screen\d+(.*)', class_name)
                if route_name_match:
                    name_part = route_name_match.group(1)
                    if not name_part:
                        route_name = class_name.lower()
                        route_path = f'/{route_name}'
                    else:
                        route_name = name_part[0].lower() + name_part[1:]
                        route_path = f'/{route_name}'
                        # Special case for splash
                        if 'splash' in route_name.lower() or class_name == 'Screen1Splash':
                            route_path = '/'
                            route_name = 'splash'
                else:
                    route_name = class_name.lower()
                    route_path = f'/{route_name}'
                    
                routes.append(f"""
      GoRoute(
        path: '{route_path}',
        name: '{route_name}',
        builder: (context, state) => const {class_name}(),
      ),""")

imports_str = '\n'.join(imports)
routes_str = ''.join(routes)

template = f"""import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:go_router/go_router.dart';

{imports_str}

final routerProvider = Provider<GoRouter>((ref) {{
  return GoRouter(
    initialLocation: '/',
    debugLogDiagnostics: true,
    routes: [{routes_str}
    ],
  );
}});
"""

with open('e:/Evoeline/lib/router/app_router.dart', 'w', encoding='utf-8') as f:
    f.write(template)

print(f"Generated {len(routes)} routes.")
